docs/sop.md

```diff
@@ -1,13 +1,95 @@
+
+**Dependency Management**: Clowder automatically manages service dependencies, ensuring that applications have access to required infrastructure services (databases, message queues, object storage) and other application services they depend on.
+
+### How Clowder Works
+
+Clowder utilizes a common configuration format that is presented to each application, no matter the environment it is running in, enabling a far easier development experience. It governs many different aspects of an application's configuration from defining the port it should listen to for its main web service, to metrics, kafka and others. When using Clowder, the burden of identifying and defining dependency and core service credentials and connection information is removed.
+
+The operator watches for changes to `ClowdApp` and `ClowdEnvironment` custom resources and automatically provisions the necessary infrastructure and generates appropriate configurations for each application.
+
+### Operating Clowder
+
+There are two primary aspects of operating Clowder: Operating the apps managed by Clowder and operating Clowder itself.
+
+## Architecture Overview
+
+### High-Level Architecture
+
+Clowder is a Kubernetes operator that manages application configuration and infrastructure dependencies for cloud-native applications. The operator follows the standard Kubernetes controller pattern, continuously reconciling desired state with actual state.
+
+#### Core Components
+
+1. **Clowder Controller Manager**
+   - Main operator process that watches for CRD changes
+   - Reconciles ClowdApp and ClowdEnvironment resources
+   - Manages application lifecycle and configuration generation
+   - Runs as a deployment in the `clowder-system` namespace
+
+2. **Custom Resource Definitions (CRDs)**
+   - `ClowdEnvironment`: Cluster-scoped resource defining infrastructure providers
+   - `ClowdApp`: Namespace-scoped resource defining application specifications
+   - `ClowdJobInvocation`: Resource for managing job executions
+
+3. **Provider System**
+   - Modular architecture supporting different infrastructure modes
+   - Providers: Database, Kafka, Object Storage, Logging, Metrics, Web, etc.
+   - Each provider supports multiple modes (local, operator, app-interface)
+
+4. **Webhooks**
+   - Validation webhooks for ClowdApp resource validation
+   - Mutation webhooks for pod injection and configuration
+
+#### Data Flow
+
+```
+ClowdApp → Controller → Provider Logic → K8s Resources → Application Config
+    ↓           ↓              ↓              ↓              ↓
+ Spec      Reconcile     Infrastructure   Deployments   cdappconfig.json
+```
+
+#### Deployment Architecture
+
+Clowder is deployed as a standard Kubernetes operator with the following components:
```

Comment:
This seems very similar to the 'core components' section above, do you feel its duplicating things?